solr/core/src/java/org/apache/solr/handler/loader/XMLLoader.java

```diff
@@ -393,31 +401,35 @@ public SolrInputDocument readDoc(XMLStreamReader parser) throws XMLStreamExcepti
             }
             if (!isLabeledChildDoc) {
               // only add data if this is not a childDoc, since it was added already
-              doc.addField(name, v);
+              doc.addField(currentFieldName, v);
             } else {
               // reset so next field is not treated as child doc
               isLabeledChildDoc = false;
             }
             // field is over
-            name = null;
+            currentFieldName = null;
           }
           break;
 
         case XMLStreamConstants.START_ELEMENT:
           text.setLength(0);
           String localName = parser.getLocalName();
           if ("doc".equals(localName)) {
-            if (name != null) {
+            if (currentFieldName != null) { // enclosed in <field>
               // flag to prevent spaces after doc from being added
               isLabeledChildDoc = true;
-              if (!doc.containsKey(name)) {
-                doc.setField(name, Lists.newArrayList());
```

Comment:
It's caused too much hard, turn back to wrapping.

Reply:
Do you want me to revert the changes that you added now?

Reply:
Hold on. I'm on it.